homeassistant/components/dsmr/config_flow.py

```diff
@@ -0,0 +1,61 @@
+"""Config flow for DSMR integration."""
+import logging
+from typing import Any, Dict, Optional
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DSMRFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for DSMR."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def _abort_if_host_port_configured(
+        self,
+        port: str,
+        host: str = None,
+        updates: Optional[Dict[Any, Any]] = None,
+        reload_on_update: bool = True,
+    ):
+        """Test if host and port are already configured."""
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
```

Comment:
`self._async_current_entries` is a helper for the current entries of the domain.